demes/demes.py

```diff
@@ -1224,20 +1242,68 @@ def simplify_migration_rates(data):
                     del migration["end_time"]
                 if migration["start_time"] == time_hi:
                     del migration["start_time"]
-            # then check for the same event in the opposite direction
-            for migration in data["migrations"]["asymmetric"]:
-                source = migration["source"]
-                dest = migration["dest"]
-                opposite = copy.deepcopy(migration)
-                opposite["dest"] = source
-                opposite["source"] = dest
-                if opposite in asymmetric:
-                    asymmetric.remove(migration)
-                    asymmetric.remove(opposite)
-                    sym = dict(demes=[dest, source], **opposite)
-                    del sym["source"]
-                    del sym["dest"]
-                    symmetric.append(sym)
+
+                k = [migration["rate"]]
+                try:
+                    k.append(migration["start_time"])
+                except KeyError:
+                    k.append(None)
```

Comment:
Here, you can use the idiom `k.append(migration.get("start_time"))` instead.

Reply:
Or indeed `k = tuple(migration.get(key) for key in ("rate", "start_time", "end_time"))`

Reply:
Oh, nice! Didn't know I could do that. Thanks